MovieTime/MovieTime/Service/MTMovieDBImageService.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MTMovieDBImageService.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 10/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import UIKit
+
+protocol MTMovieDBServiceImageProtocol: class {
+    init(service: MTServiceProtocol, helper: MTServiceHelperProtocol, config: MTModelMovieDBConfiguration)
+
+    func loadBackdropImage(filePath: String, completion: ((Data?, MTNetworkError?) -> Void)?)
+    func loadLogoImage(filePath: String, completion: ((Data?, MTNetworkError?) -> Void)?)
+    func loadPosterImage(filePath: String, completion: ((Data?, MTNetworkError?) -> Void)?)
+    func loadProfileImage(filePath: String, completion: ((Data?, MTNetworkError?) -> Void)?)
+}
+
+final class MTMovieDBImageService: MTMovieDBServiceImageProtocol {
+
+    private var configuration: MTModelMovieDBConfiguration
+    private let currentService: MTServiceProtocol
+    private let serviceHelper: MTServiceHelperProtocol
+
+    private var urlImage: URL? {
+        guard let baseImageURL = configuration.images?.base_url else { return nil }
+        return URL(string: baseImageURL)
+    }
+
+    init(service: MTServiceProtocol = MTService(), helper: MTServiceHelperProtocol = MTServiceHelper(), config: MTModelMovieDBConfiguration) {
+        currentService = service
+        serviceHelper = helper
+        configuration = config
+    }
+
+    func loadBackdropImage(filePath: String, completion: ((Data?, MTNetworkError?) -> Void)?) {
+        let fileSize = configuration.images?.backdrop_sizes?.first ?? "original"
```

Comment:
the string "original" is hardcoded several times, it would be very elegant to have all the sizes in an enum and try to initialize from a rawValue - when failed, you nil-coalesce to original

```
enum ImageSize: String {
  case something
  case original
}
let fileSize = ImageSize(rawValue: ...) ?? .original
```

Reply:
if you have an enum, you can also remove all these methods and have only one where the `ImageSize` is a parameter. The only difference between them is the image size!